apollo-router/tests/integration/subscriptions/ws_passthrough.rs

```diff
@@ -544,3 +560,208 @@ async fn test_subscription_ws_passthrough_pure_error_payload_with_coprocessor()
 
     Ok(())
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_subscription_ws_passthrough_on_config_reload() -> Result<(), BoxError> {
+    if !graph_os_enabled() {
+        eprintln!("test skipped");
+        return Ok(());
+    }
+
+    // Create fixed payloads for consistent testing
+    let custom_payloads = vec![create_user_data_payload(1), create_user_data_payload(2)];
+    let interval_ms = 10;
+
+    // Start subscription server with fixed payloads, but do not terminate the connection
+    let (ws_addr, http_server) =
+        start_subscription_server_with_payloads(custom_payloads.clone(), interval_ms, false).await;
+
+    // Create router with port reservations
+    let mut router = IntegrationTest::builder()
+        .supergraph("tests/integration/subscriptions/fixtures/supergraph.graphql")
+        .config(include_str!(
+            "fixtures/subscription_schema_reload.router.yaml"
+        ))
+        .build()
+        .await;
+
+    // Configure URLs using the string replacement method
+    let ws_url = format!("ws://{}/ws", ws_addr);
+    router.replace_config_string("http://localhost:{{PRODUCTS_PORT}}", &http_server.uri());
+    router.replace_config_string("http://localhost:{{ACCOUNTS_PORT}}", &ws_url);
+    router.replace_config_string("rng:", "accounts:");
+
+    info!("WebSocket server started at: {}", ws_url);
+
+    router.start().await;
+    router.assert_started().await;
+
+    // Use the configured query that matches our server configuration
+    let query = create_sub_query(interval_ms, custom_payloads.len());
+    let (_, response) = router.run_subscription(&query).await;
+
+    // Expect the router to handle the subscription successfully
+    assert!(
+        response.status().is_success(),
+        "Subscription request failed with status: {}",
+        response.status()
+    );
+
+    let stream = response.bytes_stream();
+    let expected_events = vec![
+        create_initial_empty_response(),
+        create_expected_user_payload(1),
+        create_expected_user_payload(2),
+        create_expected_reload_payload(),
+    ];
+
+    // try to reload the config file
+    router.replace_config_string("replaceable", "replaced");
```

Comment:
should we also do a config change related to subscriptions?

Reply:
No, it's not related to subscriptions, because we're re-using the execution pipeline we can't only reload when it's only subscription config, because it includes all the other plugins so we have to reload everytime the config changes

Reply:
gotcha!